agent/agent_worker.go

```diff
@@ -165,15 +166,34 @@ func (a *AgentWorker) Start() error {
 		a.logger.Debug("[DisconnectionTimer] Started for %d seconds...", a.agentConfiguration.DisconnectAfterJobTimeout)
 	}
 
+	a.idleMonitor = idle
+
 	// Setup an idle timer to disconnect after periods of idleness
 	if a.agentConfiguration.DisconnectAfterIdleTimeout > 0 {
 		a.idleTimer = time.NewTimer(time.Second * time.Duration(a.agentConfiguration.DisconnectAfterIdleTimeout))
 		go func() {
 			for {
 				select {
 				case <-a.idleTimer.C:
-					a.logger.Info("Agent has been idle for %d seconds", a.agentConfiguration.DisconnectAfterIdleTimeout)
-					a.stopIfIdle()
+					// Mark this agent as idle in the shared idle monitor
+					a.idleMonitor.MarkIdle(a.agent.Name)
```

Comment:
Should we use it's UUID instead of it's name? Just in case multiple agents have the same one?

Reply:
Yeah, probably a good idea.

Reply:
Done ⚡️